plutus-tx-plugin/test/IsData/9.6/deconstructData.pir.golden

```diff
@@ -85,121 +31,141 @@ let
+                                              Maybe_match
+                                                {b}
+                                                (`$dFromData` ds)
+                                                {all dead. Maybe (Tuple2 a b)}
+                                                (\(arg : b) ->
+                                                   /\dead ->
+                                                     Just
+                                                       {Tuple2 a b}
+                                                       (Tuple2 {a} {b} arg arg))
+                                                (/\dead -> Nothing {Tuple2 a b})
+                                                {all dead. dead})
+                                         (/\dead -> Nothing {Tuple2 a b})
+                                         {all dead. dead})
+                                  (/\dead -> Nothing {Tuple2 a b})
+                                  {all dead. dead}))
+                      (/\dead -> Nothing {Tuple2 a b})
+                      {all dead. dead})
+                 (/\dead -> Nothing {Tuple2 a b})
+                 {all dead. dead})
+            (\(ds : Unit) -> Nothing {Tuple2 a b})
+            (\(ds : Unit) -> Nothing {Tuple2 a b})
+            (\(ds : Unit) -> Nothing {Tuple2 a b})
+            (\(ds : Unit) -> Nothing {Tuple2 a b})
+            Unit
   ~`$fFromDataTuple2` :
      all a b.
        (\a -> data -> Maybe a) a ->
        (\a -> data -> Maybe a) b ->
        (\a -> data -> Maybe a) (Tuple2 a b)
     = `$fFromDataTuple2_$cfromBuiltinData`
+  !unsafeDataAsB : data -> bytestring = unBData
+  !unsafeDataAsI : data -> integer = unIData
+  !unsafeDataAsList : data -> list data = unListData
+  !unsafeDataAsMap : data -> list (pair data data) = unMapData
   ~`$fFromDataInteger_$cfromBuiltinData` : data -> Maybe integer
     = \(d : data) ->
         let
           !d : data = d
         in
-        matchData'
-          {Maybe integer}
+        chooseData
+          {Unit -> Maybe integer}
           d
-          (\(ds : integer) (ds : list data) -> Nothing {integer})
-          (\(ds : list (pair data data)) -> Nothing {integer})
-          (\(ds : list data) -> Nothing {integer})
-          (\(i : integer) -> let !i : integer = i in Just {integer} i)
-          (\(ds : bytestring) -> Nothing {integer})
+          (\(ds : Unit) ->
+             let
+               !tup : pair integer (list data) = unsafeDataAsConstr d
+               !ds : integer = fst {integer} {list data} tup
+               !ds : list data = snd {integer} {list data} tup
+             in
+             Nothing {integer})
+          (\(ds : Unit) ->
+             let
+               !ds : list (pair data data) = unsafeDataAsMap d
+             in
+             Nothing {integer})
+          (\(ds : Unit) ->
+             let
+               !ds : list data = unsafeDataAsList d
+             in
+             Nothing {integer})
```

Comment:
Good thing about pattern matching builtins is that they make this not only faster, but also quite shorter.